enhancements/monitoring/hypershift-monitoring.md

```diff
@@ -0,0 +1,390 @@
+
+Observability Product will be a central pane of view of observability data for SRE needs from multiple clusters, as well as alerting and alert routing for notification purposes.
+
+2. Since the Management cluster is like any other OCP/OSD, we can leverage Platform Monitoring. In near future, we plan to allow [MonitoringStack](https://github.com/openshift/enhancements/pull/866) project to allow scraping platform related metrics, but for the time being, this is done via Platform Monitoring that can remote write relevant metrics to Observability Product.
+
+3. [MonitoringStack](https://github.com/openshift/enhancements/pull/866) project offer similar functionality as UWM, but it aims for more features, flexibility and scalability due to different support models (outside of OpenShift Core) and less tight integration to OpenShift itself. It’s being developed in https://github.com/rhobs/monitoring-stack-operator. We plan to add support to [the agent mode of Prometheus](https://twitter.com/bwplotka/status/1454146102240944134) for efficient metric forwarding.
+4. A similar pipeline can be used for a portion of HSRE owned components living in each data plane.
+
+#### Customer Monitoring on Data Plane
+
+As per goals and non-goals we aim to give the CUS ability to scrape data related to their worklodads / hardware / Kubernetes state metrics that is on their data plane. As mentioned in [Non Goals](#non-goals) we don’t want to give or sell anything for scalable, centralized monitoring to customers (yet). Not beyond anything else w provide which is an opinionated UWM, CMO can deploy on normal OCP.
+
+What we have to do, is to enable and recommend them using existing workflows, which simply means providing the same data using the same protocols (Prometheus exposition format).
+
+The typical workflows:
+* HPA/VPA using Prometheus Adapter
+* Monitoring and alerting on kube-state metrics, node exporter, cadvisor and application metrics
+* Local console access with OCP metrics and dashboard
+
+We don’t see any blockers for the proposed model if we would follow the same path as GKE, EKS or any other Managed Kubernetes with hidden control plane offers. By design, HyperShift offers managed Kube API. This means that any metrics related to the things running in the data plane and around API can be obtained by Customers on their own. It could work like this:
+
+![h](assets/hypershift-cus.png)
+
+This is not different to GKE/EKS cases, where, the customer has to deploy on their own kube-state-metrics, node-exporter, cadvisor etc and configure Open Metrics compatible solution to scrape those.
+
+As the Red Hat data plane, we used to offer this out-of-the-box ,so we should maintain this practice. In simple form we could use CMO without PM, but with all exporters and optional UWM. There is a slight tweak where UWM has to know scrape those data-plane KSM, node exporter, cadvisor.
```

Comment:
3 options are presented here:
- Use CMO without PM
- Use Monitoring Stacks Operator
- Allow customer to install DIY Prometheus

Which are we recommending? Are we offering all 3?
For option #1, use CMO, there are changes that need to happen to the CMO to do this. Can we add more detail on what that is?